src/fsharp/ast.fs

```diff
@@ -1742,7 +1755,7 @@ let mkSynDotBrackSeqSliceGet  m mDot arr (argslist:list<SynIndexerArg>) =
                        | SynIndexerArg.One x -> yield x
                        | _ -> () ]
     if notsliced.Length = argslist.Length then
-        SynExpr.DotIndexedGet(arr,[SynIndexerArg.One (SynExpr.Tuple(notsliced,[],unionRanges (Seq.head notsliced).Range (Seq.last notsliced).Range))],mDot,m)
```

Comment:
Honestly I don't care much which things get proto-built. This is just one more example of how the OSS build diverges from the Microsoft build, which causes tremendous confusion and complexity for me and @KevinRansom.

Just to get it down "on paper" while I'm thinking about it, the two biggest pain points are (indirectly related to this PR):

- Proto
  - OSS build uses proto FSharp.Compiler, fsc, **FSharp.Build**
  - Msft build uses proto FSharp.Compiler, fsc, **FSharp.Core**
- LKG
  - OSS will use RTM 3.0 or 3.1 F# compiler/targets as the LKG if it finds them in the regular place
  - OSS will use the *4.0.30319.1* LKG otherwise, and for powerpack tools
  - Msft build always uses LKG for both compiler and powerpack, version *2.0.50726.900*

Just these two primary differences cause frequent issues where stuff will build for OSS, but not internally, or vice versa, and it's not trivial to work through the resolution.  I don't think anyone is against simplifying/unifying, but the whole system is so fragile that we are wont to make sweeping changes for fear of silently retargeting something the wrong way.

There are also numerous simpler issues (different NUnit version for tests, different ways of defining InternalsVisibleTo and versions, different hierarchy of projects, slightly different sets of #defines, different set of target platforms, etc) which don't make it any easier.

And now this week we are trying to accommodate requests to make the whole thing work for both VS 2013 and VS 2015 at the same time.

Sorry for the rant. I know @KevinRansom is just trying to get everything working. I'm just really trying to avoid introducing yet more complexity and divergence in the build/test infra.